packages/lesswrong/components/votes/lwReactions/NamesAttachedReactionsVoteOnComment.tsx

```diff
@@ -474,20 +445,22 @@ const ReactionOverview = ({voteProps, classes}: {
       <h3>Reacts Overview</h3>
       <div className={classes.alreadyUsedReactions}>
         {alreadyUsedReactionTypesByKarma.map(r => {
-          const usersWhoReacted = alreadyUsedReactions[r]!;
+          const reactions = alreadyUsedReactions[r]!;
```

Comment:
Are all of these instances of extra `!` usage preparing for a world where we enable `noUncheckedIndexedAccess` in tsconfig?  I think it's pretty unlikely that we'll ever enable that.

(If it's instead intended to communicate something like "the type system already says this, but it's overconfident.  In this case, I, the developer, am additionally confirming that this should never result in pulling out something that doesn't exist" then I'm not sure I'm sold that it's worth adding the additional pattern to the codebase, especially one that doesn't actually cause any changes in compiler behavior.  Would rather leave a stock comment in such cases.)

Reply:
The type system doesn't actually know this; without the `!` it thinks that `reactions` could be undefined (it can't be, for reasons the type system can't see), which causes a type error later in the function.